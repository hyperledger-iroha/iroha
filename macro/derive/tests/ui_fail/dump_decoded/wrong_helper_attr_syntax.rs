use parity_scale_codec::Decode;

#[derive(Debug, Decode, iroha_derive::DumpDecoded)]
#[dump_decoded(name = AnotherStruct)]
struct Struct;

#[derive(Debug, Decode, iroha_derive::DumpDecoded)]
#[dump_decoded(rename = "AnotherStruct2")]
struct Struct2;
