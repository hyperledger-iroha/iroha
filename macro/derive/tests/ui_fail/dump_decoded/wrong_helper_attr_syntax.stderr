error: proc-macro derive panicked
 --> tests/ui_fail/dump_decoded/wrong_helper_attr_syntax.rs:3:25
  |
3 | #[derive(Debug, Decode, iroha_derive::DumpDecoded)]
  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Invalid attribute syntax. Supported: #[dump_decoded(<attr> = "<value>")]

error: proc-macro derive panicked
 --> tests/ui_fail/dump_decoded/wrong_helper_attr_syntax.rs:7:25
  |
7 | #[derive(Debug, Decode, iroha_derive::DumpDecoded)]
  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Only `name` attribute is supported

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui_fail/dump_decoded/wrong_helper_attr_syntax.rs:1:1
  |
1 | / use parity_scale_codec::Decode;
2 | |
3 | | #[derive(Debug, Decode, iroha_derive::DumpDecoded)]
4 | | #[dump_decoded(name = AnotherStruct)]
... |
8 | | #[dump_decoded(rename = "AnotherStruct2")]
9 | | struct Struct2;
  | |_______________^ consider adding a `main` function to `$DIR/tests/ui_fail/dump_decoded/wrong_helper_attr_syntax.rs`
